src/main/java/io/cdap/plugin/gcp/gcs/sink/GCSBatchSink.java

```diff
@@ -340,6 +379,70 @@ public void validate(FailureCollector collector) {
       }
     }
 
+    public void validateContentType(FailureCollector failureCollector) {
+      switch (format) {
```

Comment:
format can be null if its a macro. I believe null with switch case will not work.

Reply:
We were able to handle null exceptions when macro is null in **GCSBatchSink** but in **GCSMultiBatchSink** it seems like there's a scenario when format is required even after validation and when the format is macro (null) this always fails with null pointer exception. 

Please check the line 75 in this file: 
**src/main/java/io/cdap/plugin/gcp/gcs/sink/GCSMultiBatchSink.java.**

```
  public void configurePipeline(PipelineConfigurer pipelineConfigurer) {
    FailureCollector collector = pipelineConfigurer.getStageConfigurer().getFailureCollector();
    config.validate(collector);
    collector.getOrThrowException();

    FileFormat format = config.getFormat();
    // add schema as a macro since we don't know it until runtime
    PluginProperties formatProperties = PluginProperties.builder()
      .addAll(config.getProperties().getProperties())
      .add("schema", String.format("${%s}", SCHEMA_MACRO)).build();
```

Reply:
That indicates a bug. Can you file a jira for it?